c_common/front_end_common_lib/include/spinn_extra.h

```diff
@@ -0,0 +1,1439 @@
+    uint : 6;
+    uint start : 1;             //!< start temperature measurement
+} sc_temperature_t;
+
+//! System controller mutex/interlock
+typedef struct {
+    uint : 31;
+    uint bit : 1;               //!< The only relevant bit in the word
+} sc_mutex_bit_t;
+
+//! System controller router control
+typedef struct {
+    uint rx_disable : 6;        //!< disables the corresponding link receiver
+    uint : 2;
+    uint tx_disable : 6;        //!< disables the corresponding link transmitter
+    uint : 2;
+    uint parity_control : 1;    //!< Router parity control
+    uint : 3;
+    uint security_code : 12;    //!< ::SYSTEM_CONTROLLER_MAGIC_NUMBER to enable write
+} sc_link_disable_t;
+
+//! System controller registers
+typedef struct {
+    const uint chip_id;         //!< Chip ID register (hardwired)
+    //! Each bit disables a processor
+    sc_magic_proc_map_t processor_disable;
+    //! Writing a 1 sets a processor’s interrupt line
+    sc_magic_proc_map_t set_cpu_irq;
+    //! Writing a 1 clears a processor’s interrupt line
+    sc_magic_proc_map_t clear_cpu_irq;
+    uint set_cpu_ok;            //!< Writing a 1 sets a CPU OK bit
+    uint clear_cpu_ok;          //!< Writing a 1 clears a CPU OK bit
+    sc_magic_proc_map_t cpu_soft_reset_level;       //!< Level control of CPU resets
+    sc_magic_proc_map_t cpu_hard_reset_level;       //!< Level control of CPU node resets
+    sc_magic_subsystem_map_t subsystem_reset_level; //!< Level control of subsystem resets
+    sc_magic_proc_map_t cpu_soft_reset_pulse;       //!< Pulse control of CPU resets
+    sc_magic_proc_map_t cpu_hard_reset_pulse;       //!< Pulse control of CPU node resets
+    sc_magic_subsystem_map_t subsystem_reset_pulse; //!< Pulse control of subsystem resets
+    const sc_reset_code_t reset_code;   //!< Indicates cause of last chip reset
+    sc_monitor_id_t monitor_id; //!< ID of Monitor Processor
+    sc_misc_control_t misc_control;     //!< Miscellaneous control bits
+    sc_io_t gpio_pull_up_down_enable;   //!< General-purpose IO pull up/down enable
+    sc_io_t io_port;            //!< I/O pin output register
+    sc_io_t io_direction;       //!< External I/O pin is input (1) or output (0)
+    sc_io_t io_set;             //!< Writing a 1 sets IO register bit
+    sc_io_t io_clear;           //!< Writing a 1 clears IO register bit
+    sc_pll_control_t pll1_freq_control; //!< PLL1 frequency control
+    sc_pll_control_t pll2_freq_control; //!< PLL2 frequency control
+    uint set_flags;             //!< Set flags register
+    uint reset_flags;           //!< Reset flags register
+    sc_clock_mux_t clock_mux_control;   //!< Clock multiplexer controls
+    const sc_sleep_status_t cpu_sleep;  //!< CPU sleep (awaiting interrupt) status
+    sc_temperature_t temperature[3];    //!< Temperature sensor registers [2:0]
+    const uint _padding[3];
+    //! Read sensitive semaphores to determine MP
+    const sc_mutex_bit_t monitor_arbiter[32];
```

Comment:
No, because they're not related. There's no reason for the number of arbiter bits to be the same as the number of test-and-set and test-and-clear registers. (The latter two are linked because of the way the system controller works.)